lib/api/unittest/CBoostedTreeInferenceModelBuilderTest.cc

```diff
@@ -135,8 +138,12 @@ BOOST_AUTO_TEST_CASE(testIntegrationRegression) {
     TStrVecVec categoryMappingVector{{}, {"cat1", "cat2", "cat3"}, {}};
     auto definition = analysisRunner->inferenceModelDefinition(fieldNames, categoryMappingVector);
 
-    LOG_DEBUG(<< "Inference model definition: " << definition->jsonString());
-    std::string modelSizeDefinition{definition->sizeInfo()->jsonString()};
+    auto modelSizeDefinition{definition->sizeInfo()->jsonString()};
+    auto definitionJsonString{definition->jsonString()};
```

Comment:
For the time being it's best to use `auto` with `=` rather than braced initialization.  So either:

```
    auto modelSizeDefinition = definition->sizeInfo()->jsonString();
    auto definitionJsonString = definition->jsonString();
```

or:

```
    std::string modelSizeDefinition{definition->sizeInfo()->jsonString()};
    std::string definitionJsonString{definition->jsonString()};
```

The meaning of `auto` with braces changed between C++14 and C++17, but different compilers implement different parts of the standards at different times so it leaves us unclear about what exactly the code is doing on different branches and different platforms.